File: Easy/Array/two-sum.cpp
// Two Sum
// Difficulty: Easy  |  Topic: Array
// https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> ans;
        unordered_map<int, int> map;

        for(int i=0; i<nums.size(); i++){
            int complement = target - nums[i];
            if(map.find(complement) != map.end()){
                ans.push_back(map[complement]);
                ans.push_back(i);
                break;
            }
            map[nums[i]] = i;
        }
        return ans;
    }
};

File: Easy/Array/two-sum.py
// Two Sum
// Difficulty: Easy  |  Topic: Array
// https://leetcode.com/problems/two-sum/

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        self.nums  = nums
        self.target = target
        rtype = []
        for i in range(len(self.nums)):
            for j in range(i+1,len(self.nums)):
                if self.nums[i]+self.nums[j]==self.target:
                    rtype.append(i)
                    rtype.append(j)
                    break
        return(rtype)